backend/app/models/field_type.rb

```diff
@@ -0,0 +1,104 @@
+# field_types table
+class FieldType < ActiveRecord::Base
+
+  # Create a field type
+  #
+  # @param field_type [Hash] the field type
+  # @option field_type[:name] [String] name
+  # @option field_type[:ftype] [String]  ftype
+  # @option field_type[:required] [String] required - interpreted as Boolen
+  # @option field_type[:array] [String] array - interpreted as Boolean
+  # @option field_type[:choices] [String] choices
+  # @option field_type[:allowable_field_types] [Array] array of allowable field types (for ftype = "sample")
+  # @return true
+  def self.create_sampletype(parent_id, field_type)
+    # Read the parameters
+    fname     = Input.text(field_type[:name])
+    ftype     = Input.text(field_type[:ftype])
+    frequired = Input.boolean(field_type[:required]) ? 1 : nil
+    farray    = Input.boolean(field_type[:array]) ? 1 : nil
+    fchoices  = Input.text(field_type[:choices])
+
+    field_type_new  = FieldType.new(
+      parent_id: parent_id,
+      name: fname,
+      ftype: ftype,
+      choices: fchoices,
+      array: farray,
+      required: frequired,
+      parent_class: "SampleType"
+    )
+    field_type_new.save
+
+    # Save allowable field types if the field type is "sample"
+    if ftype == "sample" and field_type[:allowable_field_types].kind_of?(Array)
+      field_type[:allowable_field_types].each do |aft|
+        AllowableFieldType.create(field_type_new.id, aft)
+      end
+    end
+
+    return true
+  end
+
+  # Update or create a new field type (if the id does not exist)
+  #
+  # @param feild_type = [Hash] field type
+  # @option feild_type[:id] id [Int] the id of the existing field type
+  # @option feild_type[:name] [String] name
+  # @option feild_type[:ftype] [String]  ftype
+  # @option feild_type[:required] [String] required - interpreted as Boolean
+  # @option feild_type[:array] [String] array - interpreted as Boolean
+  # @option feild_type[:choices] [String] choices
+  # @option feild_type[:allowable_field_types] [Array] array of allowable field types (for ftype = "sample")
+  # @return the id of the field type or 0 (if the id does not exist and the name is blank)
+  def self.update_sampletype(parent_id, field_type)
```

Comment:
same as above

also, field is misspelled in the block comment

Reply:
Agreed as well.